src/main/java/org/kohsuke/groovy/sandbox/SandboxTransformer.java

```diff
@@ -498,10 +498,13 @@ private Expression innerTransform(Expression exp) {
                     }
                     boolean old = visitingClosureBody;
                     visitingClosureBody = true;
+                    ClassNode oldMethodReturnType = methodReturnType;
+                    methodReturnType = ClassHelper.OBJECT_TYPE;
```

Comment:
The declared type of the generated method is always `Object.class`, see [here](https://github.com/apache/groovy/blob/41b990d0a20e442f29247f0e04cbed900f3dcad4/src/main/org/codehaus/groovy/classgen/asm/ClosureWriter.java#L219) and [here](https://github.com/apache/groovy/blob/41b990d0a20e442f29247f0e04cbed900f3dcad4/src/main/org/codehaus/groovy/classgen/asm/ClosureWriter.java#L240).